### PR TITLE
OPENNLP-1664: Switch to pre-trained UD models in WSD component

### PR DESCRIPTION
**Pull request by mawiesne**
Change
-
- switches to ud-models in opennlp-wsd component
- improves IO handling of resources in tests and ITs uses thread-safe impls of Tokenizer, POSTagger and Lemmatizer in WSDHelper 
- raises accuracy of eval results significantly (by ud-models / proper lemmatization) 
- introduces slf4j logging instead of sysout garbage 
- clears some ArrayList bloating in method signatures 
- kicks out all binary SF models from test/resources/models 
- removes dependency towards 'commons-lang3'
- introduces Disambiguator interface as common basis for the three existing impls 
- adds new JUnit tests classes, contributing to **OPENNLP-827** 
- improves JavaDoc along the path, contributing to **OPENNLP-789**

Tasks
-
Thank you for contributing to Apache OpenNLP.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with OPENNLP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn clean install` at the root opennlp-sandbox folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file in opennlp-sandbox folder?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found in opennlp-sandbox folder?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/AbstractWSDisambiguator.java [hunk 1]
```diff
@@ -22,37 +22,41 @@
 import java.security.InvalidParameterException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Pattern;
+
+import net.sf.extjwnl.JWNLException;
```
Comment (rzo1) [inline opennlp-wsd/src/main/java/opennlp/tools/disambiguator/AbstractWSDisambiguator.java:27]: Just checked the license, which is BSD-3 -> ok for us.

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/AbstractWSDisambiguator.java [hunk 2]
```diff
@@ -140,14 +126,65 @@ public List<String> disambiguate(String[] tokenizedContext,
...
+   * @param wordTag A combination of word and POS tag, separated by a {@code .} character.
+   * @return The disambiguated sense and key if disambiguation was successful,
+   *         {@code null} otherwise.
    */
-  public abstract String disambiguate(WSDSample sample);
+  protected String disambiguate(String wordTag) {
+
+    String[] splitWordTag = SPLIT.split(wordTag);
+
+    String word = splitWordTag[0];
```
Comment (rzo1) [inline opennlp-wsd/src/main/java/opennlp/tools/disambiguator/AbstractWSDisambiguator.java:144]: Is it guaranteed that `wordTag != null` and `array.length == 2` ? (most likely CnP via pull method up)

**Review by rzo1 — APPROVED** [review]

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/AbstractWSDisambiguator.java (hunk 1 above)
Comment (mawiesne) [reply opennlp-wsd/src/main/java/opennlp/tools/disambiguator/AbstractWSDisambiguator.java:27]: Thanks for checking & feedback.

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/AbstractWSDisambiguator.java (hunk 2 above)
Comment (mawiesne) [reply opennlp-wsd/src/main/java/opennlp/tools/disambiguator/AbstractWSDisambiguator.java:144]: This part can be improved in a follow-up PR as in opennl-tools exists a class WordTag which handles that better as in this CnP variant.

**Pull request merged**